src/CMakeLists.txt

```diff
@@ -1,127 +1,126 @@
-        modulefinder/sentry_modulefinder_windows.c
-    )
+	list(APPEND SENTRY_SOURCES
+		modulefinder/sentry_modulefinder_windows.c
+	)
 elseif(APPLE)
-    list(APPEND SENTRY_SOURCES
-        modulefinder/sentry_modulefinder_apple.c
-    )
+	list(APPEND SENTRY_SOURCES
+		modulefinder/sentry_modulefinder_apple.c
+	)
 elseif(LINUX OR ANDROID)
-    list(APPEND SENTRY_SOURCES
-        modulefinder/sentry_modulefinder_linux.c
-    )
+	list(APPEND SENTRY_SOURCES
+		modulefinder/sentry_modulefinder_linux.c
+	)
 endif()
 
 # curl transport
-if(WITH_CURL)
-    list(APPEND SENTRY_SOURCES
-        transports/sentry_libcurl_transport.c
-        transports/sentry_libcurl_transport.h
-    )
+if(SENTRY_CURL_SUPPORT)
+	list(APPEND SENTRY_SOURCES
+		transports/sentry_libcurl_transport.c
+		transports/sentry_libcurl_transport.h
+	)
 endif()
 if(WIN32)
-    list(APPEND SENTRY_SOURCES
-        transports/sentry_winhttp_transport.c
-        transports/sentry_winhttp_transport.h
-    )
+	list(APPEND SENTRY_SOURCES
+		transports/sentry_winhttp_transport.c
+		transports/sentry_winhttp_transport.h
+	)
 endif()
 
 # backends
 if(BACKEND_CRASHPAD)
-    list(APPEND SENTRY_SOURCES
-        backends/sentry_crashpad_backend.cpp
-        backends/sentry_crashpad_backend.h
-    )
+	list(APPEND SENTRY_SOURCES
+		backends/sentry_crashpad_backend.cpp
+		backends/sentry_crashpad_backend.h
+	)
 elseif(BACKEND_INPROC)
-    list(APPEND SENTRY_SOURCES
-        backends/sentry_inproc_backend.c
-        backends/sentry_inproc_backend.h
-    )
+	list(APPEND SENTRY_SOURCES
+		backends/sentry_inproc_backend.c
+		backends/sentry_inproc_backend.h
+	)
 endif()
 
 # unwinder
 if(WITH_LIBBACKTRACE)
-    target_compile_definitions(sentry PUBLIC SENTRY_WITH_UNWINDER_LIBBACKTRACE)
-    list(APPEND SENTRY_SOURCES
-        unwinder/sentry_unwinder_libbacktrace.c
-    )
+	target_compile_definitions(sentry PUBLIC SENTRY_WITH_UNWINDER_LIBBACKTRACE)
+	list(APPEND SENTRY_SOURCES
+		unwinder/sentry_unwinder_libbacktrace.c
+	)
 endif()
 if(WITH_LIBUNWINDSTACK)
-    target_compile_definitions(sentry PUBLIC SENTRY_WITH_UNWINDER_LIBUNWINDSTACK)
-    list(APPEND SENTRY_SOURCES
-        unwinder/sentry_unwinder_libunwindstack.cpp
-    )
+	target_compile_definitions(sentry PUBLIC SENTRY_WITH_UNWINDER_LIBUNWINDSTACK)
+	list(APPEND SENTRY_SOURCES
+		unwinder/sentry_unwinder_libunwindstack.cpp
+	)
 endif()
 if(WIN32)
-    target_compile_definitions(sentry PUBLIC SENTRY_WITH_UNWINDER_DBGHELP)
-    list(APPEND SENTRY_SOURCES
-        unwinder/sentry_unwinder_dbghelp.c
-    )
+	target_compile_definitions(sentry PUBLIC SENTRY_WITH_UNWINDER_DBGHELP)
+	list(APPEND SENTRY_SOURCES
+		unwinder/sentry_unwinder_dbghelp.c
+	)
 endif()
 
 # make paths absolute
 PREPEND(SENTRY_SOURCES "${CMAKE_CURRENT_SOURCE_DIR}/" ${SENTRY_SOURCES})
-get_target_property(SENTRY_OLD_SOURCES sentry SOURCES)
-target_sources(sentry PRIVATE ${SENTRY_OLD_SOURCES} ${SENTRY_SOURCES})
+target_sources(sentry PRIVATE ${SENTRY_SOURCES})
```

Comment:
does this work correctly?
I had the problem that just using `target_sources` overrides the `vendor/mpack.c` set in the main cmake file, thats why I did this dance here. But CI is passing, so I hope everything is correct?

Reply:
`target_sources` only adds sources (as a set), it does not remove.
All sources are now added as an absolute path, so `vendor/mpack.c` cannot be overridden.

I can imagine the problem you're describing happening if there are 2 `vendor/mpack.c` sources in different folders, and the sources are added as a relative path. But that is not the case here.